test/testutil.c

```diff
@@ -400,18 +412,17 @@ int test_mem_eq(const char *file, int line, const char *st1, const char *st2,
 
     if (s1 == NULL && s2 == NULL)
         return 1;
-    if (n1 != n2) {
-        test_fail_message(NULL, file, line, "memory",
-                          "size mismatch %s %s [%"OSSLzu"] != %s %s [%"OSSLzu"]",
-                          st1, print_mem_maybe_null(s1, n1, b1), n1,
-                          st2, print_mem_maybe_null(s2, n2, b2), n2);
-        return 0;
-    }
-    if (s1 == NULL || s2 == NULL || memcmp(s1, s2, n1) != 0) {
+    if (n1 != n2 || s1 == NULL || s2 == NULL || memcmp(s1, s2, n1) != 0) {
+        char *m1 = print_mem_maybe_null(s1, n1, b1);
+        char *m2 = print_mem_maybe_null(s2, n2, b2);
+
         test_fail_message(NULL, file, line, "memory",
-                          "%s %s [%"OSSLzu"] != %s %s [%"OSSLzu"]",
-                          st1, print_mem_maybe_null(s1, n1, b1), n1,
-                          st2, print_mem_maybe_null(s2, n2, b2), n2);
+                          "%s %s [%"OSSLzu"] == %s %s [%"OSSLzu"]",
```

Comment:
The way these messages are read depends on the audience. They are obviously meant to be read the same way you would read an assertion failure, i.e. "the expression `blah` failed". This is probably obvious to programmers, but I can certainly see how it can be read differently, and I understand this is where @dot-asm is going with his comment? 

As far as this particular line goes, `==` seems perfectly correct since the expression being tested here is equally (as long as we stay with "the expression `blah` failed" kind of output)